tests/cache/tests.py

```diff
@@ -1174,11 +1175,56 @@ def test_createcachetable_with_table_argument(self):
         out = io.StringIO()
         management.call_command(
             'createcachetable',
-            'test cache table',
+            self.table_name,
             verbosity=2,
             stdout=out,
         )
-        self.assertEqual(out.getvalue(), "Cache table 'test cache table' created.\n")
+        self.assertEqual(out.getvalue(), f"Cache table '{self.table_name}' created.\n")
+
+    def test_quoted_query_has_key(self):
+        """Tests that the query values are quoted in DatabaseCache.has_key().
+        Refs #33340 (regression test)
+        """
+        key = 'test_has_key'
+        now = timezone.now().replace(microsecond=0, tzinfo=None)
+        with mock.patch(
+            'django.db.backends.utils.CursorWrapper.execute',
+        ) as mocked_cursor_execute:
+            cache.has_key(key)
+            mocked_cursor_execute.assert_called_with(
+                'SELECT %s FROM %s WHERE %s = %%s and %s > %%s' % (
+                    connection.ops.quote_name('cache_key'),
+                    connection.ops.quote_name(self.table_name),
+                    connection.ops.quote_name('cache_key'),
+                    connection.ops.quote_name('expires')
+                ),
+                [f":1:{key}", connection.ops.adapt_datetimefield_value(now)]
+            )
```

Comment:
There is not need to wrap `CursorWrapper.execute`, we should be able to use `CaptureQueriesContext()`, e.g.
```python
    def test_has_key_query_columns_quoted(self):
        with CaptureQueriesContext(connection) as captured_queries:
            cache.has_key('key')
        self.assertEqual(len(captured_queries), 1)
        sql = captured_queries[0]['sql']
        # Column names are quoted.
        self.assertIn(connection.ops.quote_name('expires'), sql)
        self.assertIn(connection.ops.quote_name('cache_key'), sql)
```
There is also no need to check a table name because it contains spaces, so it's already tested.

Reply:
Done ☑️